pkg/api/version/version.go

```diff
@@ -0,0 +1,279 @@
+package version
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/bufbuild/connect-go"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/gogo/protobuf/proto"
```

Comment:
I'm not very sure if we want to use "gogoproto" instead of `google.golang.org/protobuf/proto`.

Also, AFAIK, the official `proto` does not rely on former `proto.Unmarshaler` and `proto.Marshaler` which where deprecated

Reply:
This is forced by the dskit/memberlist package and the mergeable I had to implement

Reply:
We could implement own [`Codec`](https://github.com/grafana/dskit/blob/main/kv/codec/codec.go#L9-L16) that uses `google.golang.org/protobuf/proto`. Either way, I don't feel strong about it, we can change this later if we want